http-server-netty/src/main/java/io/micronaut/http/server/netty/handler/Compressor.java

```diff
@@ -52,6 +57,8 @@ final class Compressor {
     private final SnappyOptions snappyOptions;
 
     Compressor(HttpCompressionStrategy strategy) {
+        assert strategy.isEnabled();
```

Comment:
should we raise a [ConfigurationException](https://github.com/micronaut-projects/micronaut-core/blob/4.4.x/inject/src/main/java/io/micronaut/context/exceptions/ConfigurationException.java) instead?

Reply:
it's only a sanity check, the callers do not invoke the constructor at all in this case